src/ui/components/Widgets/Chart/ChartWidget.tsx

```diff
@@ -79,13 +87,28 @@ export const ChartWidget = (props: ChartWidgetProps) => {
     const initName = React.useMemo(() => currentTab.title, [currentTab]);
     const {chartId, enableActionParams} = currentTab;
 
+    const widgetDatasetFields = React.useMemo(
+        () =>
+            (dashDatasetFields?.find((item) => item.entryId === chartId)?.datasetFields || null) as
+                | string[]
+                | null,
+        [chartId, dashDatasetFields],
+    );
+    const prevWidgetDatasetFields = usePrevious(widgetDatasetFields);
+
+    const loadedDSFields = widgetDatasetFields && widgetDatasetFields !== prevWidgetDatasetFields;
+
     const chartkitParams = React.useMemo(() => {
-        let res = removeEmptyProperties(props.params);
+        let res = removeEmptyNDatasetFieldsProperties(
```

Comment:
We need to reject extra re-render (and api re-run) because of changed dataset fields params after loading dataset fields on dash